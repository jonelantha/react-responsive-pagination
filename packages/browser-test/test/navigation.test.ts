import { setupThrowOnError } from './helper';

beforeAll(async () => {
  setupThrowOnError(page, { ignoreInvalidPropTypes: true });

  await page.goto(`${harnessUrl}bootstrap4`);

  await page.setViewportSize({ width: 500, height: 700 });
});

describe('Pagination navigation', () => {
  beforeAll(async () => {
    await page.fill('#totalAsJson', '100');
  });

  test.each([
    ['»', 2],
    ['»', 3],
    ['100', 100],
    ['«', 99],
    ['93', 93],
  ])(
    'clicking %s renders correctly and sets current page to %i',
    async (linkToClick, expectedCurrent) => {
      await page.click(`text="${linkToClick}"`);

      const paginationHtml = await page.$eval('ul.pagination', ul => ul.outerHTML);

      expect(paginationHtml).toMatchSnapshot();

      const currentInput = await page.$('#currentAsJson');
      await expect(currentInput).toEqualValue(expectedCurrent.toString());

      // url should be unchanged; don't add a # to the url after clicking
      const currentUrl = page.url();
      await expect(currentUrl).toBe(`${harnessUrl}bootstrap4`);
    },
  );

  test.each([[2], [5], [-1], [101], [null], [''], ['3']])(
    'setting current page programatically to %p renders correctly',
    async programmaticCurrent => {
      await page.fill('#currentAsJson', JSON.stringify(programmaticCurrent));

      const paginationHtml = await page.$eval('ul.pagination', ul => ul.outerHTML);
      expect(paginationHtml).toMatchSnapshot();
    },
  );
});
