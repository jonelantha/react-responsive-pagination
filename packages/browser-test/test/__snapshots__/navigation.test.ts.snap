// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination navigation clicking 93 renders correctly and sets current page to 93 1`] = `"<ul class="pagination justify-content-center"><li class="page-item"><a class="page-link" href="#" aria-label="Previous"><span aria-hidden="true">«</span><span class="sr-only">Previous</span></a></li><li class="page-item"><a class="page-link" href="#">1</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">91</a></li><li class="page-item"><a class="page-link" href="#">92</a></li><li class="page-item active"><a class="page-link" href="#" aria-label="(current)"><span aria-hidden="true">93</span><span class="sr-only">(current)</span></a></li><li class="page-item"><a class="page-link" href="#">94</a></li><li class="page-item"><a class="page-link" href="#">95</a></li><li class="page-item"><a class="page-link" href="#">96</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span><span class="sr-only">Next</span></a></li></ul>"`;

exports[`Pagination navigation clicking 100 renders correctly and sets current page to 100 1`] = `"<ul class="pagination justify-content-center"><li class="page-item"><a class="page-link" href="#" aria-label="Previous"><span aria-hidden="true">«</span><span class="sr-only">Previous</span></a></li><li class="page-item"><a class="page-link" href="#">1</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">93</a></li><li class="page-item"><a class="page-link" href="#">94</a></li><li class="page-item"><a class="page-link" href="#">95</a></li><li class="page-item"><a class="page-link" href="#">96</a></li><li class="page-item"><a class="page-link" href="#">97</a></li><li class="page-item"><a class="page-link" href="#">98</a></li><li class="page-item"><a class="page-link" href="#">99</a></li><li class="page-item active"><a class="page-link" href="#" aria-label="(current)"><span aria-hidden="true">100</span><span class="sr-only">(current)</span></a></li><li class="page-item disabled"><span class="page-link"><span aria-hidden="true">»</span><span class="sr-only">Next</span></span></li></ul>"`;

exports[`Pagination navigation clicking « renders correctly and sets current page to 99 1`] = `"<ul class="pagination justify-content-center"><li class="page-item"><a class="page-link" href="#" aria-label="Previous"><span aria-hidden="true">«</span><span class="sr-only">Previous</span></a></li><li class="page-item"><a class="page-link" href="#">1</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">93</a></li><li class="page-item"><a class="page-link" href="#">94</a></li><li class="page-item"><a class="page-link" href="#">95</a></li><li class="page-item"><a class="page-link" href="#">96</a></li><li class="page-item"><a class="page-link" href="#">97</a></li><li class="page-item"><a class="page-link" href="#">98</a></li><li class="page-item active"><a class="page-link" href="#" aria-label="(current)"><span aria-hidden="true">99</span><span class="sr-only">(current)</span></a></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span><span class="sr-only">Next</span></a></li></ul>"`;

exports[`Pagination navigation clicking » renders correctly and sets current page to 2 1`] = `"<ul class="pagination justify-content-center"><li class="page-item"><a class="page-link" href="#" aria-label="Previous"><span aria-hidden="true">«</span><span class="sr-only">Previous</span></a></li><li class="page-item"><a class="page-link" href="#">1</a></li><li class="page-item active"><a class="page-link" href="#" aria-label="(current)"><span aria-hidden="true">2</span><span class="sr-only">(current)</span></a></li><li class="page-item"><a class="page-link" href="#">3</a></li><li class="page-item"><a class="page-link" href="#">4</a></li><li class="page-item"><a class="page-link" href="#">5</a></li><li class="page-item"><a class="page-link" href="#">6</a></li><li class="page-item"><a class="page-link" href="#">7</a></li><li class="page-item"><a class="page-link" href="#">8</a></li><li class="page-item"><a class="page-link" href="#">9</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span><span class="sr-only">Next</span></a></li></ul>"`;

exports[`Pagination navigation clicking » renders correctly and sets current page to 3 1`] = `"<ul class="pagination justify-content-center"><li class="page-item"><a class="page-link" href="#" aria-label="Previous"><span aria-hidden="true">«</span><span class="sr-only">Previous</span></a></li><li class="page-item"><a class="page-link" href="#">1</a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item active"><a class="page-link" href="#" aria-label="(current)"><span aria-hidden="true">3</span><span class="sr-only">(current)</span></a></li><li class="page-item"><a class="page-link" href="#">4</a></li><li class="page-item"><a class="page-link" href="#">5</a></li><li class="page-item"><a class="page-link" href="#">6</a></li><li class="page-item"><a class="page-link" href="#">7</a></li><li class="page-item"><a class="page-link" href="#">8</a></li><li class="page-item"><a class="page-link" href="#">9</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span><span class="sr-only">Next</span></a></li></ul>"`;

exports[`Pagination navigation setting current page programatically to "" renders correctly 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link"><span aria-hidden="true">«</span><span class="sr-only">Previous</span></span></li><li class="page-item active"><a class="page-link" href="#" aria-label="(current)"><span aria-hidden="true">1</span><span class="sr-only">(current)</span></a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item"><a class="page-link" href="#">3</a></li><li class="page-item"><a class="page-link" href="#">4</a></li><li class="page-item"><a class="page-link" href="#">5</a></li><li class="page-item"><a class="page-link" href="#">6</a></li><li class="page-item"><a class="page-link" href="#">7</a></li><li class="page-item"><a class="page-link" href="#">8</a></li><li class="page-item"><a class="page-link" href="#">9</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span><span class="sr-only">Next</span></a></li></ul>"`;

exports[`Pagination navigation setting current page programatically to "3" renders correctly 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link"><span aria-hidden="true">«</span><span class="sr-only">Previous</span></span></li><li class="page-item active"><a class="page-link" href="#" aria-label="(current)"><span aria-hidden="true">1</span><span class="sr-only">(current)</span></a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item"><a class="page-link" href="#">3</a></li><li class="page-item"><a class="page-link" href="#">4</a></li><li class="page-item"><a class="page-link" href="#">5</a></li><li class="page-item"><a class="page-link" href="#">6</a></li><li class="page-item"><a class="page-link" href="#">7</a></li><li class="page-item"><a class="page-link" href="#">8</a></li><li class="page-item"><a class="page-link" href="#">9</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span><span class="sr-only">Next</span></a></li></ul>"`;

exports[`Pagination navigation setting current page programatically to -1 renders correctly 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link"><span aria-hidden="true">«</span><span class="sr-only">Previous</span></span></li><li class="page-item active"><a class="page-link" href="#" aria-label="(current)"><span aria-hidden="true">1</span><span class="sr-only">(current)</span></a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item"><a class="page-link" href="#">3</a></li><li class="page-item"><a class="page-link" href="#">4</a></li><li class="page-item"><a class="page-link" href="#">5</a></li><li class="page-item"><a class="page-link" href="#">6</a></li><li class="page-item"><a class="page-link" href="#">7</a></li><li class="page-item"><a class="page-link" href="#">8</a></li><li class="page-item"><a class="page-link" href="#">9</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span><span class="sr-only">Next</span></a></li></ul>"`;

exports[`Pagination navigation setting current page programatically to 2 renders correctly 1`] = `"<ul class="pagination justify-content-center"><li class="page-item"><a class="page-link" href="#" aria-label="Previous"><span aria-hidden="true">«</span><span class="sr-only">Previous</span></a></li><li class="page-item"><a class="page-link" href="#">1</a></li><li class="page-item active"><a class="page-link" href="#" aria-label="(current)"><span aria-hidden="true">2</span><span class="sr-only">(current)</span></a></li><li class="page-item"><a class="page-link" href="#">3</a></li><li class="page-item"><a class="page-link" href="#">4</a></li><li class="page-item"><a class="page-link" href="#">5</a></li><li class="page-item"><a class="page-link" href="#">6</a></li><li class="page-item"><a class="page-link" href="#">7</a></li><li class="page-item"><a class="page-link" href="#">8</a></li><li class="page-item"><a class="page-link" href="#">9</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span><span class="sr-only">Next</span></a></li></ul>"`;

exports[`Pagination navigation setting current page programatically to 5 renders correctly 1`] = `"<ul class="pagination justify-content-center"><li class="page-item"><a class="page-link" href="#" aria-label="Previous"><span aria-hidden="true">«</span><span class="sr-only">Previous</span></a></li><li class="page-item"><a class="page-link" href="#">1</a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item"><a class="page-link" href="#">3</a></li><li class="page-item"><a class="page-link" href="#">4</a></li><li class="page-item active"><a class="page-link" href="#" aria-label="(current)"><span aria-hidden="true">5</span><span class="sr-only">(current)</span></a></li><li class="page-item"><a class="page-link" href="#">6</a></li><li class="page-item"><a class="page-link" href="#">7</a></li><li class="page-item"><a class="page-link" href="#">8</a></li><li class="page-item"><a class="page-link" href="#">9</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span><span class="sr-only">Next</span></a></li></ul>"`;

exports[`Pagination navigation setting current page programatically to 101 renders correctly 1`] = `"<ul class="pagination justify-content-center"><li class="page-item"><a class="page-link" href="#" aria-label="Previous"><span aria-hidden="true">«</span><span class="sr-only">Previous</span></a></li><li class="page-item"><a class="page-link" href="#">1</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">93</a></li><li class="page-item"><a class="page-link" href="#">94</a></li><li class="page-item"><a class="page-link" href="#">95</a></li><li class="page-item"><a class="page-link" href="#">96</a></li><li class="page-item"><a class="page-link" href="#">97</a></li><li class="page-item"><a class="page-link" href="#">98</a></li><li class="page-item"><a class="page-link" href="#">99</a></li><li class="page-item active"><a class="page-link" href="#" aria-label="(current)"><span aria-hidden="true">100</span><span class="sr-only">(current)</span></a></li><li class="page-item disabled"><span class="page-link"><span aria-hidden="true">»</span><span class="sr-only">Next</span></span></li></ul>"`;

exports[`Pagination navigation setting current page programatically to null renders correctly 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link"><span aria-hidden="true">«</span><span class="sr-only">Previous</span></span></li><li class="page-item active"><a class="page-link" href="#" aria-label="(current)"><span aria-hidden="true">1</span><span class="sr-only">(current)</span></a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item"><a class="page-link" href="#">3</a></li><li class="page-item"><a class="page-link" href="#">4</a></li><li class="page-item"><a class="page-link" href="#">5</a></li><li class="page-item"><a class="page-link" href="#">6</a></li><li class="page-item"><a class="page-link" href="#">7</a></li><li class="page-item"><a class="page-link" href="#">8</a></li><li class="page-item"><a class="page-link" href="#">9</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span><span class="sr-only">Next</span></a></li></ul>"`;
